### PR TITLE
test(runtime-metrics): test env variable

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

Add a test case to ensure that ddtrace-run behaves as expected based on the value of the `DD_RUNTIME_METRICS_ENABLED` environment variable.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by jd — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

File: tests/runtime/test_runtime_metrics_api.py [hunk 1]
```diff
@@ -43,3 +43,28 @@ def test_manually_start_runtime_metrics(run_python_code_in_subprocess):
...
+    assert status == 0
+
+    monkeypatch.setenv("DD_RUNTIME_METRICS_ENABLED", "true")
+
+    _, _, status, _ = ddtrace_run_python_code_in_subprocess(
+        """
+from ddtrace.internal.runtime.runtime_metrics import RuntimeWorker
+assert RuntimeWorker._instance is not None
+"""
+    )
```
Comment (brettlangdon) [inline tests/runtime/test_runtime_metrics_api.py:69]:
```suggestion
""",
    env={"DD_RUNTIME_METRICS_ENABLED": "true"},
    )
```

`ddtrace_run_python_code_in_subprocess` passed extra kwargs to `subprocess.Popen` via `call_program` https://github.com/DataDog/dd-trace-py/blob/ffa3826aa9337682dc53f75852a12034b4b7f246/tests/utils.py#L947-L950

**Review by brettlangdon — APPROVED** [review]

File: tests/runtime/test_runtime_metrics_api.py [hunk 2]
```diff
@@ -43,3 +43,27 @@ def test_manually_start_runtime_metrics(run_python_code_in_subprocess):
...
+def test_start_runtime_metrics_via_env_var(ddtrace_run_python_code_in_subprocess):
+    """
+    When running with ddtrace-run and DD_RUNTIME_METRICS_ENABLED is set
+        Runtime metrics worker starts and there are no errors
+    """
+
+    _, _, status, _ = ddtrace_run_python_code_in_subprocess(
+        """
+from ddtrace.internal.runtime.runtime_metrics import RuntimeWorker
+assert RuntimeWorker._instance is None
```
Comment (Kyle-Verhoog) [inline tests/runtime/test_runtime_metrics_api.py:57]:
```suggestion
assert RuntimeWorker.status == ServiceStatus.STOPPED
```

we've been bitten by having our tests rely too much on implementation details in the past 😅

File: tests/runtime/test_runtime_metrics_api.py [hunk 3]
```diff
@@ -43,3 +43,27 @@ def test_manually_start_runtime_metrics(run_python_code_in_subprocess):
...
+"""
+    )
+    assert status == 0
+
+    _, _, status, _ = ddtrace_run_python_code_in_subprocess(
+        """
+from ddtrace.internal.runtime.runtime_metrics import RuntimeWorker
+assert RuntimeWorker._instance is not None
```
Comment (Kyle-Verhoog) [inline tests/runtime/test_runtime_metrics_api.py:65]:
```suggestion
assert RuntimeWorker.status == ServiceStatus.RUNNING
```

File: tests/runtime/test_runtime_metrics_api.py (hunk 2 above)
Comment (P403n1x87) [reply tests/runtime/test_runtime_metrics_api.py:57]: Great suggestion but for this to work we'd need to implement `status` as a class method on `RuntimeWorker`, or as a static method on the public `RuntimeMetrics` with a new PR. WDYT?

Comment (Kyle-Verhoog) [reply tests/runtime/test_runtime_metrics_api.py:57]: ahh right. Hmm I don't think it should block this PR but maybe it'd be nice to have in the future!

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**